linux/create_installer_linux.sh

```diff
@@ -0,0 +1,135 @@
+#! /bin/bash
+
+# Jenkins build script for the job create_installer_linux. It repackages all
+# tarballs named OpenJDK*.tar.gz into OS packages that have been copied into
+# the root folder of the Jenkins' workspace using Gradle. If desired, the
+# resulting packages are automatically published if they haven't been published
+# yet (see environment variable RELEASE_TYPE below).
+#
+# If you want to package AdoptOpenJDK builds yourself, use Gradle directly.
+#
+# The script expects some environment variables to be present:
+#
+#   RELEASE_TYPE          "Release", "Nightly" or "Nightly Without Publish".
+#                         See AdoptOpenJDK build pipelines for details.
+#
+#   WORKSPACE             Path to the root directory of Jenkins' workspace.
+#
+#   MAJOR_VERSION         Major version of AdoptOpenJDK, e.g. 8, 9, 11
+#
+#   VERSION               Complete version string, e.g. 1.8.0_222-b10
+#
+#   ARCHITECTURE          Name of the CPU architecture, e.g. x64. See
+#                         AdoptOpenJDK build pipelines for valid values.
+#
+#   ARTIFACTORY_PASSWORD  Password of the user that is used to push releases
+#                         to Artifactory.
+#
+# Optional variables:
+#
+#   ARTIFACTORY_USER            Name of the Artifactory user
+#
+#   ARTIFACTORY_REPOSITORY_DEB  Name of the Artifactory repository for DEB
+#                               packages.
+#
+#   ARTIFACTORY_REPOSITORY_RPM  Name of the Artifactory repository for RPM
+#                               packages.
+
+set -eu
+shopt -s globstar nullglob nocaseglob nocasematch
+
+if [ -z "${ARTIFACTORY_USER+x}" ]; then
+    ARTIFACTORY_USER="adoptopenjdk-jenkins-bot"
+fi
+
+DISTRIBUTION_DIR=$(mktemp -d -t adoptjdkbuild)
+
+finish() {
+    if [ -d "$DISTRIBUTION_DIR" ] ; then
+        rm -rf "$DISTRIBUTION_DIR"
+    fi
+}
+trap finish EXIT
+
+echoerr() { printf "%s\n" "$*" >&2; }
+
+case $RELEASE_TYPE in
```

Comment:
could this block be wrapped in a function?

Reply:
What would be the advantage? As far as I know, Bash does not really have return values and the code isn‘t re-used anywhere. Happy to make the change, but hard if I don‘t see what you are aiming at.

Reply:
In that case, we can just add a comment at the top of the block explaining what it broadly does

Reply:
Hmm, the previous comment didn't save.

In that case, perhaps add a comment at the top of the block?

Reply:
Comments have been added.